week-1/fix/median.js

```diff
@@ -3,9 +3,12 @@
 // If you're in the week-1 directory, you can run npm test -- fix to run the tests in the fix directory
 
 function calculateMedian(list) {
-  const middleIndex = Math.floor(list.length / 2);
+  // const list = [1, 2, 3];
+  const middleIndex = (list.length / 2);
   const median = list.splice(middleIndex, 1)[0];
-  return median;
+  if (list.length % 2 === 0) {
+  return median;}
+  return (middleIndex + (middleIndex + 1)) / 2; 
```

Comment:
Hi Haytham , I don't think solution is gonna work because first of all you are doing other way around in your if else statement. If you can reivew this article to understand how to get median both when the number of numbers are odd or even (https://www.mathsisfun.com/median.html)

Then after you understand the article you flip the if statement check. Also in line 11, you are doing `middleIndex + 1` this is not going to work becaues you are doing array index plus 1 instead of getting the value of the middleIndex  + 1 from array

Reply:
Also, you need to sort the array before your median, you find that part in the article I shared with you!

Reply:
Thanks, Altom for your guidance. There was a misunderstanding on my side. I thought the task was to find the element in the middle of the array. Not after sorting the array. but now I have changed the function to sort the array before finding the median element.